openedx/core/djangoapps/discussions/data.py

```diff
@@ -0,0 +1,45 @@
+"""
+Data classes for discussions
+"""
+from typing import List
+
+import attr
+from opaque_keys.edx.keys import CourseKey, UsageKey
+
+# TODO: These data models will be moved to openedx_events. They are currently here to simplify the PR.
+
+
+@attr.s(frozen=True)
+class DiscussionTopicContext:
+    """
+    Context for linking the external id for a discussion topic to its associated usage key.
+
+    The ``title`` is cached to improve the performance, since otherwise we'd
+    need to look it up in the course structure each time.
+
+    The ``group_id`` can be used for providers that don't internally support
+    cohorting but we can emulate that wuth different contexts for different groups.
+    """
+    title = attr.ib(type=str)
+    usage_key = attr.ib(type=UsageKey, default=None)
+    group_id = attr.ib(type=int, default=None)
+    external_id = attr.ib(type=str, default=None)
```

Comment:
What is this external_id used for?

Reply:
The `external_id` is what will store the commentable id that is used by `cs_comments_service`. In the future, if we are dealing with another provider that supports in-context discussions, this can be used for them as well.